wordpress-mobile/WordPress-iOS
# Jetpack Custom Icons

## Description
This PR adds the ability to set a custom icon for the Jetpack app. This feature existed in WordPress before, so the code was tweaked a bit to support it for Jetpack as well.

See Internal reference: 1203003448154690-as-1202975621869625.

## Screenshots
This PR also changes the table view style from "grouped" to "inset grouped" for both WordPress and Jetpack:

**WordPress**
| Before | After |
| ------ | ----- |
| ![](https://user-images.githubusercontent.com/9609223/193477704-3be2fa54-8d19-4e85-873c-bfecc05cc517.jpeg) | ![](https://user-images.githubusercontent.com/9609223/193477614-fe57c33c-33c3-4943-940f-2c8bfdcfda6d.png) |

**Jetpack**
| Light | Dark |
| ------ | ----- |
| ![light](https://user-images.githubusercontent.com/9609223/195108777-dc3e0dd2-66f1-4a9c-b4d1-862c5428a97a.png) | ![dark](https://user-images.githubusercontent.com/9609223/195109063-8d5c092d-a570-42d1-a7f7-f4617705ef14.png) |

## Test Instructions
Do the following for both WordPress and Jetpack:
1. Make sure you're on "My Site" tab.
2. Tap Profile Picture at the top right corner.
3. Tap "App Settings".
4. Tap "App Icon" row at the bottom.
5. Make sure the Icons screen look good in both Dark and Light modes.
6. Select any Icon.
7. Close the app, and make sure app icon is the same one you selected.

## Regression Notes
1. Potential unintended areas of impact
These changes impact the WordPress Custom Icons feature. 

2. What I did to test those areas of impact (or what existing automated tests I relied on)
I manually tested this feature for both WordPress and Jetpack apps. 

3. What automated tests I added (or what prevented me from doing so)
I added the following Unit Tests:
- Tests that the app icons list contains exactly 1 primary icon
- Tests that the app icons list contains at least 1 alternative icon

## PR submission checklist:
